src/GitVersion.Core/Configuration/GitVersionConfiguration.cs

```diff
@@ -64,18 +57,15 @@ public GitVersionConfiguration()
     [YamlMember(Alias = "tag-pre-release-weight")]
     public int? TagPreReleaseWeight { get; set; }
 
-    [YamlMember(Alias = "commit-message-incrementing")]
-    public CommitMessageIncrementMode? CommitMessageIncrementing { get; set; }
+    [YamlMember(Alias = "unknown-branch")]
+    public BranchConfiguration? UnknownBranch { get; set; }
```

Comment:
Please move this into `Branches` and call it `Unknown`.

```suggestion
```

Reply:
Just want to ask you @asbjornu and you @arturcic (and others?) to have a Microsoft teams meeting to discuss the structure of the configuration? If I read the https://github.com/GitTools/GitVersion/issues/2870 issue (and others) it gives me the encouragement that the configuration (how it is defined today) is not intuitive and self-explained. We need to discuss in my opinion the following points:

- Separate the fallback branch configuration from the general configuration
- Separate the unknown branch configuration from the normal branch configuration
- Extending the configuration from the user perspective which are already pre defined

I'm not happy with the direction where we are going with the configuration if I'm honest. I see a lot of tickets coming to us in the future because of misusage and misunderstanding. What is about clean code and self explanation? Furthermore if you are not on board with the templating idea @asbjornu what is your suggestion? Could you please elaborate how to support different workflows in future?

If it is because of backwards compatibility we could support the old structure and a new structure of the configuration and define the old one as deprecated in future releases.

Reply:
I agree that the current state of affairs in v5 is sub-optimal. By making the configuration completely explicit in every way, I think the situation will be much improved. I'm not against the templating idea for backwards compatibility concerns, but because it seems to add needless complexity. The configuration should be simple. The reason it's not simple today is not due to the lack of a template system, but because it's so hard to deduce what the effective value of a given property in the branch configuration is supposed to be at any point. If we make that explicit, we can document it and reason about it much more clearly.